pkgs/development/python-modules/python-imports-check-libstdcxx/default.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, gcc12Stdenv # The last release
+, gcc11Stdenv # Second-to-last, don't forget to bump
+, buildPythonPackage
+, python3
+, callPackage
+, psutil
+}:
+
+let
+  basic = callPackage ./basic.nix { };
+
+  # We could expose just "basic", but we want to hydra to fail if the following
+  # sanity checks do not pass:
+
+  loadOld = basic.overridePythonAttrs (a: {
+    pname = "${basic.pname}-load-old";
+    buildInputs = a.buildInputs ++ [
+      gcc11Stdenv.cc.cc.lib
+    ];
+    nativeBuildInputs = a.nativeBuildInputs ++ [
+      gcc11Stdenv.cc
+    ];
+    cmakeFlags = a.cmakeFlags ++ [
+      "-DLOAD_LIBSTDCXX_MODULE_NAME=load_libstdcxx_old"
+    ];
+    pythonImportsCheck = [
+      "load_libstdcxx_old"
+    ];
+  });
+  loadNew = basic.overridePythonAttrs (a: {
+    pname = "${basic.pname}-load-new";
+    buildInputs = a.buildInputs ++ [
+      gcc12Stdenv.cc.cc.lib
+    ];
+    nativeBuildInputs = a.nativeBuildInputs ++ [
+      gcc12Stdenv.cc
+    ];
+    cmakeFlags = a.cmakeFlags ++ [
+      "-DLOAD_LIBSTDCXX_MODULE_NAME=load_libstdcxx_new"
+    ];
+    pythonImportsCheck = [
+      "load_libstdcxx_new"
+    ];
+    EXPECTED_LIBSTDCXX = "${gcc11Stdenv.cc.cc.lib}";
```

Comment:
why does `loadNew` have `EXPECTED_LIBSTDCXX` but `loadOld` doesn't?

Reply:
Because I'm silly! And I didn't notice this because I've never actually built `loadNew`, I only used it to construct `checkFails`.

Good catch!